bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateResultInfo.java

```diff
@@ -54,6 +54,23 @@ protected FirmwareUpdateResultInfo() {
         }
     }
 
+    /**
+     * Creates a new {@link FirmwareUpdateResultInfo}.
+     *
+     * @param thingUID thingUID of the thing being updated
+     * @param result the result of the firmware update (must not be null)
+     * @param errorMessage the error message in case of result is {@link FirmwareUpdateResult#ERROR} (must not be null
+     *            or empty for erroneous firmware updates; ignored for successful firmware updates)
+     * @return FirmwareUpdateResultInfo (not null)
+     * @throws NullPointerException if the thingUID or firmware update result is null
+     * @throws IllegalArgumentException if error message is null or empty for erroneous firmware updates
+     */
+    @NonNull
```

Comment:
see https://www.eclipse.org/smarthome/documentation/development/conventions.html

Reply:
Thanks! Will be cleared out in the next PR update.